MetaMask/metamask-extension
# Fix autodetect tokens link not jumping to section in settings

## **Description**

When clicking the link to enable token autodetection in settings, you land at the top of the settings page instead of jumping to the specific setting.

Also fixes a bug where the import tokens modal did not automatically close when clicking the link. (see video)

Fixed by using the correct settings identifier and calling `onClose()`

## **Manual testing steps**

1. Open a metamask wallet 
2. Turn token autodetection off in settings
3. Click import tokens
4. Click the button to enable token autodetection in settings
5. Verify whether you land at the specific token autodetection setting

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/3500406/1d34f1c7-fe4e-43ca-9fcc-d5a2417b0dc8

### **After**

https://github.com/MetaMask/metamask-extension/assets/3500406/503c8b31-93df-4996-9003-8a53efa10699


## **Related issues**


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
